src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -270,7 +261,7 @@ bool DiscoveryDataBase::update(
 {
     if (!enabled_)
     {
-        logWarning(DISCOVERY_DATABASE, "Ignoring attempt on updating a disabled database");
+        logError(DISCOVERY_DATABASE, "Discovery Database is disabled");
```

Comment:
Warning was ok. With multiple threads it could happen, the important fact is that we do not process it if disabled.